Calculate unit summaries

software-project opened the pull request:
Issue #54

software-project commented:
There are still some things that need to be handled (like translations and currencies) but I wanted you guys to have a look first.

bk3c commented:
Another option is maybe we take the summary method as an API query param, with just the byType current one to start?

software-project commented on services/listings/src/lib/unit_transformations.ts:
```diff
@@ -1,79 +1,84 @@
 }
 
-export const transformUnitsIntoGroups = (units: Units): GroupedUnitsWithSummaries => {
+const groupUnits = (units: Units, type: keyof Unit = "unit_type"): UnitGroup[] => {
```
We have this already. Have a look on line 78. We are grouping with summaries reserved and priority. I'm open for different schema there, though.

bk3c replied:
yeah, this was more of a comment about naming than about functionality, but I'm fine merging as-is.

software-project commented:
@bk3c I addressed issues that you wanted (expect snake vs camel case as you made a separate ticket for it). Have a look on my grouping comment and on `UnitSummarized` interface

bk3c commented on shared/styles/tailwind.config.js:
```diff
@@ -88,16 +90,15 @@ module.exports = {
-      normal: '0',
-      wide: '.025em',
-      wider: '.05em',
-      widest: '.1em',
```
@slowbot Note widest was defined twice here, so I just guessed which one was correct so it would pass linting. Feel free to fix it as you see fit.

bk3c commented:
Ok, I feel like there's still too much duplication in the current JSON format, but I'm going to merge this since it's directionally correct and it's a stable base we can move forward from.
